src/nomad_simulations/model_system.py

```diff
@@ -283,12 +318,216 @@ def normalize(self, archive, logger) -> None:
         super().normalize(archive, logger)
 
 
+def convert_combo_to_name(combo: tuple[str]) -> str:
+    """Convert a combo tuple to a hypen-separated string.
+    For angles, the center element is now indeed placed at the center."""
+    if 1 < len(combo) < 4:
+        if len(combo) == 3:
+            combo = (combo[1], combo[0], combo[2])  # flip order
+        return '-'.join(combo)
+    raise ValueError('Only pairs and triplets are supported.')
+
+
+class GeometryDistribution(ArchiveSection):
```

Comment:
> Instead of a repeating subsection of the same class type, I think this might need to be changed to a container section which may hold any geometrical representation.

Okay... What would be the defining feature of the container section then? That it's a histogram? I'm fine looking into defining this with you, but this sounds as it should be a follow-up issue. **I can open up an issue.**

> These subsections should then inherit from PhysicalProperty for consistency purposes

I'm not really sure about the approach behind `PhysicalProperty`, but it sounds contradicting to the multiple subsections statement. I would go either with a _(object) compositional approach_ or a _type system_, but not mix them.

Reply:
That's fine, then please open an issue for later. But yes, the general structure is a histogram.

I don't see the contradiction. I am saying that GeometricDistribution() should (perhaps) inherit from PhysicalProperty(), and then the individual classes inherit from GeometricDistribution() as below. Am I missing something? (Anyway, we should also deal with this later)

Reply:
So it sounds like a _type_ approach, sure. How on point is `PhysicalProperty`, though? I'll open an issue after this one.